gtsam_unstable/slam/ProjectionFactorRollingShutter.h

```diff
@@ -0,0 +1,257 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file ProjectionFactorRollingShutter.h
+ * @brief Basic projection factor for rolling shutter cameras
+ * @author Yotam Stern
+ */
+
+#pragma once
+
+#include <gtsam/nonlinear/NonlinearFactor.h>
+#include <gtsam/geometry/PinholeCamera.h>
+#include <gtsam/geometry/CalibratedCamera.h>
+#include <gtsam/geometry/Cal3_S2.h>
+#include <boost/optional.hpp>
+
+namespace gtsam {
+
+/**
+ * Non-linear factor for 2D projection measurement obtained using a rolling shutter camera. The calibration is known here.
+ * This version takes rolling shutter information into account as follows: consider two consecutive poses A and B,
+ * and a Point2 measurement taken starting at time A using a rolling shutter camera.
+ * Pose A has timestamp t_A, and Pose B has timestamp t_B. The Point2 measurement has timestamp t_p (with t_A <= t_p <= t_B)
+ * corresponding to the time of exposure of the row of the image the pixel belongs to.
+ * Let us define the interp_param = (t_p - t_A) / (t_B - t_A), we will use the pose interpolated between A and B by
```

Comment:
Not loving `interp_param`. Can we just call this `alpha`? Also note in docs that alpha is between 0.0 and 1.0.

Reply:
done